sqlglot/optimizer.py

```diff
@@ -0,0 +1,27 @@
+import sqlglot.expressions as exp
+
+
+def qualify_tables(expression, db=None, catalog=None):
+    def qualify(node):
+        if isinstance(node, exp.Table):
+            if not node.args.get("db"):
+                node.set("db", db)
+            if not node.args.get("catalog"):
+                node.set("catalog", catalog)
+            return node
+        return node
+
+    return expression.transform(qualify)
+
+
+def qualify_columns(expression, _schema):
+    return expression
```

Comment:
What were you imaging here?
I was thinking DFS in postorder, disambiguating columns as we go up the tree?